Changes to start 'libtoolizing' CFITSIO, made with the utilities and …

bryanirby opened the pull request:
…m4 files from glibtoolize 2.4.7, aclocal 1.16.5, automake 1.16.5, autoconf 2.72

Further changes are needed, including handling of shared library version numbers

bryanirby committed:
Changes to start 'libtoolizing' CFITSIO, made with the utilities and m4 files from glibtoolize 2.4.7, aclocal 1.16.5, automake 1.16.5, autoconf 2.72

Further changes are needed, including handling of shared library version numbers

lpsinger reviewed (commented):
I haven't tested this yet, but from a quick read-through it looks really good.

Rather than put the files generated by autoconf under version control, what if you _excluded_ them from version control and had your GitHub Actions workflow run autoreconf etc. in order to make release tarballs?

lpsinger commented on Makefile.am:
```diff
@@ -0,0 +1,182 @@
+F77_WRAPPERS = f77_wrap1.c f77_wrap2.c f77_wrap3.c f77_wrap4.c
+else
+F77_WRAPPERS = 
+endif
```
Careful, this will cause the source tarball to be lacking these source files if you configure without Fortran present. Instead, I suggest the following idiom:

```
F77_WRAPPERS = f77_wrap1.c f77_wrap2.c f77_wrap3.c f77_wrap4.c

...
libcfitsio_la_SOURCES = ...
...

if !NOFORTRAN
libcfitsio_la_SOURCES += $(F77_WRAPPERS)
endif
 
...

EXTRA_DIST = ... $(F77_WRAPPERS)
```

lpsinger reviewed (commented):
The part of configure.ac labeled `Shared library section` is the part that is problematic for linking against libcfitsio as a shared library. Would you please remove that and let libtool do its magic?

bryanirby committed:
Reworked handling of F77_WRAPPERS and GSIFTP_SRC to make sure they're included in the distribution (issue reported by L.Singer via GitHub).

bryanirby committed:
Removed old shared library section to prevent pollution of libtool environment; also, fixed Fortran wrapper switch.

bryanirby committed:
Added placeholder files required by the libtool/automake system.

The GNU file "COPYING" is (I believe) superseded by licenses/NASA_Open_Source_Agreement_1.3.txt.

bryanirby committed:
Added install hook to create shared library symbolic links, attempting to maintain backward compatibility

bryanirby committed:
Fixed version number substitution in cfitsio.pc.

bryanirby committed:
Fixes for GSIFTP conditional, cfitsioLibObjs target, and handling of Fortran for testf77 program

lpsinger commented on configure.ac:
```diff
@@ -233,42 +263,38 @@ fi
+
+# Use GNU Fortran wrappers by default:
+
+if test "x$use_fortran" = "xyes"; then
+  AC_MSG_NOTICE(cfitsio: == Adding Fortran wrappers)
+  #AC_PROG_FC
+  if test "x$FC" != "x" ; then F77=$FC; fi
+  AC_PROG_F77
+  #AC_CHECK_PROGS(FC, gfortran g95 g77 f77 ifort f95 f90 xlf cf77 gf77 af77 ncf f2c, notfound)
+  if test "x$F77" != "x" ; then
+     #FC_BASENAME=`basename $FC`
+     FC_BASENAME=`basename $F77`
+     if test "x$FC_BASENAME" = "xf2c" ; then
+        AC_MSG_NOTICE(cfitsio: == Adding wrapper support for f2c)
```
What about using one of these built-in autoconf macros to detect the Fortran compiler? https://www.gnu.org/software/autoconf/manual/autoconf-2.67/html_node/Fortran-Compiler.html

bryanirby replied:
It is using AC_PROG_F77 - are you thinking it should be using a different macro?

lpsinger replied:
Oh, no, I just didn't notice that. I think I was looking at AC_CHECK_PROGS and didn't realize that it was commented out.

lpsinger reviewed (commented):
What is this meant to do?

lpsinger commented on COPYING:
This is supposed to be your license file, I think.

bryanirby replied:
Yes, there's some debate about our current license, so I'm waiting to see how that plays out.

lpsinger commented on Makefile.am:
```diff
@@ -0,0 +1,199 @@
+
+libcfitsio_ladir = $(pkgincludedir)
+libcfitsio_la_HEADERS =	fitsio.h fitsio2.h longnam.h drvrsmem.h \
+	cfortran.h f77_wrap.h region.h
```
This causes the headers to be installed into `${prefix}/include/cfitsio`, whereas before they were just installed to `${prefix}/include`. For backward compatibility, I suggest sticking with the old location:

```suggestion
include_HEADERS = fitsio.h fitsio2.h longnam.h drvrsmem.h \
	cfortran.h f77_wrap.h region.h
```

If you decide to keep the new location, then please make sure to update cfitsio.pc.in.

bryanirby replied:
Thanks for catching that, I prefer to retain backward compatibility.

lpsinger replied:
Would you please apply that change so that I can test it again?

bryanirby committed:
Update Makefile.am for backward compatibility of header file location (include/ rather than include/cfitsio)

Co-authored-by: Leo Singer <leo.p.singer@nasa.gov>

bryanirby committed:
Regenerated from Makefile.am using autoreconf(automake).
